Fix minigame bugs and refactor.

jessetilro opened the pull request:
Important, fixes bugs multiple minigames. Removes unnecessary components.

Commit:
Refactor client reorganize packages and layouts.

Refactored the client. Reorganized the packages so the main package only
contains activities. Removed unnecessary components. Switched the
minigames for the engineer and the scientist to conform to our design
document.

Commit:
Fix AsteroidImpact minigame.

The AsteroidImpact minigame now returns to role activity after completing
the event.

Commit:
Fix PlasmaLeak minigame.

Fixed the PlasmaLeak minigame to not send a failure message, validation is
done exclusively on server-side now.

FlXlT commented on app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/minigame/AsteroidImpactActivity.java:
```diff
@@ -61,16 +64,17 @@ public void createProgress(View view) {
-        }
 
-        if (progressBar.getProgress() != 100) {
-            progressBar.incrementProgressBy(5);
```
I'm not sure why people touched this, because now you can keep caling the method to increment, and I did it in this order (first checking for 95, to achieve consistency).

It should be `progressBar.getProcess() == 95` so that no multiple messages can  be send...

jessetilro replied:
The activity is stopped when going over 95,  so no double messages are sent. Please trust me, it works perfectly this way, even better than before. Otherwise the player would have kept hanging around in the completed minigame, so he/she has no feedback the minigame was completed.

dalderliesten commented:
Looks good to me after you check Floris' comment.

dmalarev commented on app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationEngineroomActivity.java:
```diff
@@ -8,9 +8,9 @@
 import nl.tudelft.pixelperfect.event.type.EventTypes;
-import nl.tudelft.pixelperfect.pixelperfect.mini_game.PlasmaLeakActivity;
+import nl.tudelft.pixelperfect.pixelperfect.minigame.AsteroidImpactActivity;
+import nl.tudelft.pixelperfect.pixelperfect.minigame.PlasmaLeakActivity;
```
i think this import can be removed, since it's no longer used in this class

dmalarev commented on app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationLabActivity.java:
```diff
@@ -9,8 +9,8 @@
 import nl.tudelft.pixelperfect.pixelperfect.R;
-import nl.tudelft.pixelperfect.pixelperfect.Spaceship;
-import nl.tudelft.pixelperfect.pixelperfect.mini_game.AsteroidImpactActivity;
+import nl.tudelft.pixelperfect.pixelperfect.minigame.AsteroidImpactActivity;
```
same goes for this import

dmalarev commented on app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/minigame/PlasmaLeakActivity.java:
```diff
@@ -143,9 +130,6 @@ public void complete(View view) {
             game.sendMessage(message);
             Intent intent = new Intent(this, LocationEngineroomActivity.class);
```
This needs to be changed so it goes to the right location

dmalarev commented:
Rest looks good

FlXlT commented:
LGTM

Commit:
Fix feedback.

Fixed Dmitry's feedback on the pull-request: remove unused imports and fix
location activity that plasma leak minigame goes back to.

jessetilro commented:
@dmalarev Thanks, fixed all your comments. Is it ok now?

dmalarev commented:
LGTM
